src/api/get-shapes.ts

```diff
@@ -0,0 +1,63 @@
+import axios from "axios";
+import fs from "fs";
+import tmp from "tmp";
+import { rhubarbCmd } from "../utils";
+import { exec } from "child_process";
+
+interface AudioAndShapes {
+  shapesFileName: string;
+  audioFileName: string;
+  shapesID: string;
+  valid: boolean;
+}
+
+const shapesID = (path: string): string => {
+  const parts: Array<string> = path.split("/");
+  const fileName: string = parts[parts.length - 1];
+  const things: Array<string> = fileName.split(".");
+  return things[0];
+};
+
+const downloadAudio = async (request: string): Promise<string | boolean> => {
+  const audioFileName = tmp.tmpNameSync({ prefix: "voice_", postfix: ".wav" });
+  const response = await axios
+    .get(request, { responseType: "arraybuffer" })
+    .catch(function(error) {
+      console.log("Error", error.message);
+    });
+  if (response) {
+    fs.writeFileSync(audioFileName, Buffer.from(response.data, "binary"));
+    return audioFileName;
+  } else {
+    return false;
+  }
+};
+
+const getShapes = async (request: string): Promise<AudioAndShapes> => {
+  const audioFileName = await downloadAudio(request);
+  const shapesFileName = tmp.tmpNameSync({
+    prefix: "shapes_",
+    postfix: ".txt"
+  });
+  const fileNames: AudioAndShapes = {
+    audioFileName: String(audioFileName),
+    shapesFileName: audioFileName ? shapesFileName : String(false),
+    shapesID: shapesID(shapesFileName),
+    valid: audioFileName ? true : false
+  };
+  if (!fileNames.valid) {
+    return fileNames;
+  }
+  const cmd = `${rhubarbCmd} -r phonetic -f json -o ${fileNames.shapesFileName} ${fileNames.audioFileName}`;
+  console.info(`cmd: ${cmd}`);
```

Comment:
Is this a standard way to log for a Node app?

Reply:
Yep. It will be captured in any logs on the place it is running.